src/analysis/convex/between.lean

```diff
@@ -487,6 +522,81 @@ end
 
 end strict_ordered_comm_ring
 
+section linear_ordered_ring
+
+variables [linear_ordered_ring R] [add_comm_group V] [module R V] [add_torsor V P]
+
+local attribute [instance] linear_ordered_ring.decidable_lt
```

Comment:
I don't understand why this is necessary.

I presume it cannot be replaced with `classical` in the proof (since you have not done this) but I would prefer to begin the proof with:
```lean
letI : decidable_rel ((<) : R → R → Prop) := linear_ordered_ring.decidable_lt,
```
than to set this local instance.

Reply:
It's the same issue with comments in two different places on such local instances in `data.sign`: linear orders are meant to be decidable, but Lean seems unable to find the decidability instances in some cases. `classical` indeed doesn't work, because it gives the wrong instance for use of `sign` lemmas in the proof (and if I replace the first two `exact` that fail with `convert`, the use of `sign_sum` then runs into such problems), so I've changed to `letI` as requested, with a comment pointing to the `data.sign` comments.